### PR TITLE
DEV-8666: addition of swap, fx-forward instrument creation tests

### PR DESCRIPTION
**Pull request by ppwfinbourne**
The SDK generation creates various instruments using some of the newer polymorphic features in open swagger.
These test performs some verification that the SDK generation has not lost these.

# Pull Request Checklist

- [ ] Read the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using Lusid.Sdk.Model;
+using NUnit.Framework;
+
+namespace Lusid.Sdk.Tests.Tutorials.Misc
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs:6]: I think we should put these in a new Tutorials/Instruments folder so its easy for users to find

File: sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                fgnAmount: usdJpyFxRate,
+                domCcy: "USD",
+                fgnCcy: "JPY",
+                startDate: new DateTimeOffset(2020, 2, 7, 0, 0, 0, TimeSpan.Zero),
+                maturityDate: new DateTimeOffset(2020, 9, 18, 0, 0, 0, TimeSpan.Zero)
+                );
+
+            // ASSERT that it was created
+            Assert.That(fxForward, Is.Not.Null);
+            // CAN NOW UPSERT TO LUSID
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs:27]: is there a reason we're not doing the upsert here (and below)?

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/ComplexInstruments.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+using Lusid.Sdk.Api;
+using Lusid.Sdk.Model;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+
+namespace Lusid.Sdk.Tests.Tutorials.Misc
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/ComplexInstruments.cs:8]: Can you update the namespace to Lusid.Sdk.Tests.Tutorials.Instruments

File: sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs (hunk 2 above)
Comment (ppwfinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs]: I didn't think anybody would need it. Have changed to have that.

File: sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs (hunk 1 above)
Comment (ppwfinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Misc/ComplexInstruments.cs]: ok

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/ComplexInstruments.cs (hunk 3 above)
Comment (ppwfinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/ComplexInstruments.cs]: yes, done.

**Pull request merged**